storage-node-v10-ha-ra-grs-v3/README.md

```diff
@@ -0,0 +1,93 @@
+---
+page_type: sample
+languages:
+- javascript
+- nodejs
+products:
+- azure
+- azure-storage
+description: "This sample shows how to use the Node.js V10 Storage SDK with read-access geo-redundant storage (RA-GRS)."
+---
+
+# Build a highly available app in Node.js with RA-GRS storage
+
+This sample shows how to use the Node.js V10 Storage SDK with read-access geo-redundant storage (RA-GRS) to create a highly available application that accesses files from secondary storage when there is a problem with primary storage, and then switches back when primary storage becomes available again. For more information, see [Designing HA Apps with RA-GRS storage](https://docs.microsoft.com/azure/storage/common/storage-designing-ha-apps-with-ragrs).
+
+If you don't have a Microsoft Azure subscription, you can get a FREE trial account <a href="http://go.microsoft.com/fwlink/?LinkId=330212">here</a>.
+
+## How it works
+
+This sample creates a new container in blob storage and uploads a sample file into it called `HelloWorld.jpg`. It then checks secondary storage repeatedly until both the container and file have been replicated. The user can then enter D to download the file or Q to quit. If the file is successfully downloaded, the sample indicates whether it came from primary or secondary storage. When the user quits the sample, it deletes the container and file. 
+
+## How to run the sample
+
+To run this sample and verify that files are downloaded from primary storage:
+
+1. Add your storage account credentials to the `.env.example` file and then rename it to `.env`.
+
+    ```
+    AZURE_STORAGE_ACCOUNT_NAME=<replace with your storage account name>
+    AZURE_STORAGE_ACCOUNT_ACCESS_KEY=<replace with your storage account access key>
+    ```
+
+    You can find this information in the Azure portal by navigating to your storage account and selecting **Access keys** in the **Settings** section. 
+
+2. Install the required dependencies in the console window by entering `npm install`.
+
+3. Launch the sample from the console window by entering `node index.js`.
+
+4. Wait until the sample reports that the container and file have replicated to secondary storage.
+
+5. When prompted, press D to download the sample file and verify that it comes from primary storage.
+
+To test secondary storage access:
```

Comment:
Tell the user what we are going to do before you ask them to execute the steps.  

To test secondary storage access, we are going to...